src/hotspot/share/opto/vectornode.cpp

```diff
@@ -43,186 +43,142 @@ int VectorNode::opcode(int sopc, BasicType bt) {
-    default: ShouldNotReachHere(); return 0;
+    default:      return 0;
     }
   case Op_AbsL:
-    assert(bt == T_LONG, "must be");
-    return Op_AbsVL;
+    return (bt == T_LONG ? Op_AbsVL : 0);
   case Op_MinI:
     switch (bt) {
     case T_BOOLEAN:
     case T_CHAR:   return 0;
     case T_BYTE:
     case T_SHORT:
     case T_INT:    return Op_MinV;
-    default:       ShouldNotReachHere(); return 0;
+    default:       return 0;
     }
   case Op_MinL:
-    assert(bt == T_LONG, "must be");
-    return Op_MinV;
+    return (bt == T_LONG ? Op_MinV : 0);
   case Op_MinF:
-    assert(bt == T_FLOAT, "must be");
-    return Op_MinV;
+    return (bt == T_FLOAT ? Op_MinV : 0);
   case Op_MinD:
-    assert(bt == T_DOUBLE, "must be");
-    return Op_MinV;
+    return (bt == T_DOUBLE ? Op_MinV : 0);
   case Op_MaxI:
     switch (bt) {
     case T_BOOLEAN:
     case T_CHAR:   return 0;
     case T_BYTE:
     case T_SHORT:
     case T_INT:    return Op_MaxV;
-    default:       ShouldNotReachHere(); return 0;
+    default:       return 0;
     }
   case Op_MaxL:
-    assert(bt == T_LONG, "must be");
-    return Op_MaxV;
+    return (bt == T_LONG ? Op_MaxV : 0);
   case Op_MaxF:
-    assert(bt == T_FLOAT, "must be");
-    return Op_MaxV;
+    return (bt == T_FLOAT ? Op_MaxV : 0);
   case Op_MaxD:
-    assert(bt == T_DOUBLE, "must be");
-    return Op_MaxV;
+    return (bt == T_DOUBLE ? Op_MaxV : 0);
   case Op_AbsF:
-    assert(bt == T_FLOAT, "must be");
-    return Op_AbsVF;
+    return (bt == T_FLOAT ? Op_AbsVF : 0);
   case Op_AbsD:
-    assert(bt == T_DOUBLE, "must be");
-    return Op_AbsVD;
+    return (bt == T_DOUBLE ? Op_AbsVD : 0);
   case Op_NegI:
-    assert(bt == T_INT, "must be");
-    return Op_NegVI;
+    return (bt == T_INT ? Op_NegVI : 0);
   case Op_NegF:
-    assert(bt == T_FLOAT, "must be");
-    return Op_NegVF;
+    return (bt == T_FLOAT ? Op_NegVF : 0);
   case Op_NegD:
-    assert(bt == T_DOUBLE, "must be");
-    return Op_NegVD;
+    return (bt == T_DOUBLE ? Op_NegVD : 0);
   case Op_RoundDoubleMode:
-    assert(bt == T_DOUBLE, "must be");
-    return Op_RoundDoubleModeV;
+    return (bt == T_DOUBLE ? Op_RoundDoubleModeV : 0);
   case Op_RotateLeft:
-    assert(bt == T_LONG || bt == T_INT, "must be");
-    return Op_RotateLeftV;
+    return (bt == T_LONG || bt == T_INT ? Op_RotateLeftV : 0);
   case Op_RotateRight:
-    assert(bt == T_LONG || bt == T_INT, "must be");
-    return Op_RotateRightV;
+    return (bt == T_LONG || bt == T_INT ? Op_RotateRightV : 0);
   case Op_SqrtF:
-    assert(bt == T_FLOAT, "must be");
-    return Op_SqrtVF;
+    return (bt == T_FLOAT ? Op_SqrtVF : 0);
   case Op_SqrtD:
-    assert(bt == T_DOUBLE, "must be");
-    return Op_SqrtVD;
+    return (bt == T_DOUBLE ? Op_SqrtVD : 0);
   case Op_PopCountI:
-    if (bt == T_INT) {
-      return Op_PopCountVI;
-    }
     // Unimplemented for subword types since bit count changes
     // depending on size of lane (and sign bit).
-    return 0;
```

Comment:
Above comment should be changes.

Reply:
Do you think so? It is related to `PopCountI`, precedes `(bt == T_INT)` check and clearly states it's about subword types.

Reply:
Okay. Yes, it is still valid.